plugins/woocommerce-admin/client/products/fields/category-field/create-category-modal.tsx

```diff
@@ -0,0 +1,192 @@
+import { Button, Modal, Spinner, TextControl } from '@wordpress/components';
+import { useMemo, useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import {
+	__experimentalSelectControl as SelectControl,
+	__experimentalSelectControlMenu as Menu,
+} from '@woocommerce/components';
+import { recordEvent } from '@woocommerce/tracks';
+import {
+	EXPERIMENTAL_PRODUCT_CATEGORIES_STORE_NAME,
+	ProductCategory,
+} from '@woocommerce/data';
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import './create-category-modal.scss';
+import { useCategorySearch } from './use-category-search';
+import { CategoryFieldItem } from './category-field-item';
+
+type CreateCategoryModalProps = {
+	initialCategoryName?: string;
+	onCancel: () => void;
+	onCreated: ( newCategory: ProductCategory ) => void;
+};
+
+export const CreateCategoryModal: React.FC< CreateCategoryModalProps > = ( {
+	initialCategoryName,
+	onCancel,
+	onCreated,
+} ) => {
+	const {
+		categoriesSelectList,
+		isSearching,
+		categoryTreeKeyValues,
+		searchCategories,
+		getFilteredItems,
+	} = useCategorySearch();
+	const { createNotice } = useDispatch( 'core/notices' );
+	const [ isCreating, setIsCreating ] = useState( false );
+	const { createProductCategory, invalidateResolutionForStoreSelector } =
+		useDispatch( EXPERIMENTAL_PRODUCT_CATEGORIES_STORE_NAME );
+	const [ categoryName, setCategoryName ] = useState(
+		initialCategoryName || ''
+	);
+	const [ categoryParent, setCategoryParent ] = useState< Pick<
+		ProductCategory,
+		'id' | 'name'
+	> | null >( null );
+
+	const onSave = async () => {
+		recordEvent( 'product_category_add', {
+			new_product_page: true,
+		} );
+		setIsCreating( true );
+		try {
+			const newCategory: ProductCategory = await createProductCategory( {
+				name: categoryName,
+				parent: categoryParent ? categoryParent.id : undefined,
+			} );
+			invalidateResolutionForStoreSelector( 'getProductCategories' );
+			setIsCreating( false );
+			onCreated( newCategory );
+		} catch ( e ) {
+			createNotice(
+				'error',
+				__( 'Failed to create category.', 'woocommerce' )
+			);
+			setIsCreating( false );
+			onCancel();
+		}
+	};
+
+	const onInputChange = ( searchString?: string ) => {
+		searchCategories( searchString || '' );
+	};
+
+	const searchDelayed = useMemo(
+		() => debounce( onInputChange, 150 ),
+		[ onInputChange ]
+	);
+
+	return (
+		<Modal
+			title={ __( 'Create category', 'woocommerce' ) }
+			onRequestClose={ () => onCancel() }
+			className="woocommerce-create-new-category-modal"
+		>
+			<div className="woocommerce-create-new-category-modal__wrapper">
+				<TextControl
+					label={ __( 'Name', 'woocommerce' ) }
+					name="Tops"
+					value={ categoryName }
+					onChange={ setCategoryName }
+				/>
+				<SelectControl< Pick< ProductCategory, 'id' | 'name' > >
+					items={ categoriesSelectList }
+					label={ __( 'Parent category (optional)', 'woocommerce' ) }
```

Comment:
Made a similar comment [here](https://github.com/woocommerce/woocommerce/pull/35266#discussion_r1003839238) but we might consider moving out `(optional)` to make it easier on translators.